index.js

```diff
@@ -47,51 +47,247 @@ const player = {
+}
+
+function searchByQueryInPlaylist(query) {
+  return searchByQueryInArr(query, player.playlists).sort(compareFunction('name'))
+}
+
+function searchByQueryInArr(query, arr) {
+  let matchArr = []
+  for (let obj of arr) {
+    for (let property of Object.values(obj)) {
+      if (('' + property).toLowerCase().includes(query)) {
+        matchArr.push(obj)
+        break
+      }
+    }
+  }
+  return matchArr
+}
+
+function compareFunction(byProperty) {
+  // returns a compareFunction based on what property to sort by, e.g "title" or "name".
+  // remember: we are sorting objects.
+  return (a, b) => {
+    if (a[byProperty] < b[byProperty]) {
+      return -1
+    }
+    if (a[byProperty] > b[byProperty]) {
+      return 1
+    }
+    return 0
+  }
+}
+
+function getClosestSong(duration) {
+  /* In this function we compare the distance between the given
+  duration and the songs duration. every time, we keep the closest song */
+  let closestSong = player.songs[0]
+  for (let song of player.songs) {
+    if (Math.abs(closestSong.duration - duration) >= Math.abs(song.duration - duration)) {
+      closestSong = song
+    }
+  }
+  return closestSong
+}
+
+function getClosestPlaylist(duration) {
+  /* Same as in getClosestSong but this time, to get the playlist
+  duration, we use the function playlistDuration */
+  let closestPlaylist = player.playlists[0]
+  for (let playlist of player.playlists) {
+    if (
+      Math.abs(playlistDuration(closestPlaylist.id) - duration) >=
+      Math.abs(playlistDuration(playlist.id) - duration)
+    ) {
+      closestPlaylist = playlist
+    }
+  }
+  return closestPlaylist
+}
+
+// end of help functions. ---------
+
 function playSong(id) {
-  // your code here
+  player.playSong(getSong(id))
 }
 
 function removeSong(id) {
-  // your code here
+  removeSongFromSongs(id)
+  removeSongFromPlaylists(id)
 }
 
-function addSong(title, album, artist, duration, id) {
-  // your code here
+function addSong(title,album,artist,duration,id = generateNewIdInArrayOfObjects(player.songs)) {
+  
+  checkIfObjectIdTaken(id, getSong)
+  console.log('test');
+  console.log(duration);
+  player.songs.push({
+    title,
+    album,
+    artist,
+    duration:convertMinFormatToSec(duration),
+    id,
+  })
+  return id
 }
 
 function removePlaylist(id) {
-  // your code here
+  removeObjectFromArray(id, player.playlists)
 }
 
-function createPlaylist(name, id) {
-  // your code here
+function createPlaylist(name, id = generateNewIdInArrayOfObjects(player.playlists)) {
+  checkIfObjectIdTaken(id, getPlaylist)
+
+  player.playlists.push({ id, name, songs: [] })
+  return id
 }
 
 function playPlaylist(id) {
-  // your code here
+  getPlaylist(id).songs.forEach((songId) => playSong(songId))
 }
 
 function editPlaylist(playlistId, songId) {
-  // your code here
+  // <getPlaylist and getSong> checks if such objects exist.
+  const SongsArr = getPlaylist(playlistId).songs
```

Comment:
The convention for variable names is `camelBack`, so the first letter should be lowercase.
And anyway you call simply call this `playlistSongs` or just `songs`